sklearn/datasets/_samples_generator.py

```diff
@@ -220,25 +243,28 @@ def make_classification(
         )
 
     if weights is not None:
+        # we define new variable, weight_, instead of modifying user defined parameter.
         if len(weights) not in [n_classes, n_classes - 1]:
             raise ValueError(
                 "Weights specified but incompatible with number of classes."
             )
         if len(weights) == n_classes - 1:
             if isinstance(weights, list):
-                weights = weights + [1.0 - sum(weights)]
```

Comment:
this isn't modifying the existing variable, it's allocating a new chunk of memory, and the name`weights` refers to that new chunk. So the original data passed to this function is never changed. Therefore this change in this PR is unnecessary.

Reply:
I did not intend to modify the variable `weights` passed by the user this is why I created new variable `weights_`